### PR TITLE
[FEAT][PGIO-114] - Analytics - track important events

### PR DESCRIPTION
**Pull request by ElRodrigote**
## Description
* Removes deprecated options in the Fathom load fn (see [this link](https://github.com/derrickreimer/fathom-client?tab=readme-ov-file#allowed-domains))
* Creates a SSOT to manage our FA event labeling
* Implements event tracking for interactions in/with:
  * Navbar "Get sDAI" button triggering the LiFi Widget 
  * Markets list: 
    * categories
    * search bar
    * filters
    * specific market selection
  * Market details:
    * details tabs and inner tabs selection
    * news article selection
    * info external links clicks to Reality, Gnosis Scan and Kleros
* [Pending]: 
  * [x] footer
  * [x] embed markets
  * [x] redeems
  * [x] assess any other pending important event. 

## Visual Evidence
Screenshot 2024-10-25 at 13 51 46

### CODE REVIEW

File: app/(main)/markets/News.tsx [hunk 1]
```diff
@@ -77,6 +80,7 @@ const NewsArticle = ({ url, title }: NewsArticleProps) => {
       href={url}
       target="_blank"
       rel="noopener noreferrer"
+      onClick={() => trackEvent(FA_EVENTS.MARKETS.DETAILS.TABS.NEWS_ARTICLE(id))}
```
Comment (Diogomartf) [inline app/(main)/markets/News.tsx:83]: I think here URL would be nicer

File: constants/index.ts [hunk 2]
```diff
@@ -76,3 +76,61 @@ export const DISCORD_URL =
   process.env.NEXT_PUBLIC_DISCORD_URL || 'https://discord.com/invite/QFkNsjTkzD';
 export const GITHUB_URL =
   process.env.NEXT_PUBLIC_GITHUB_URL || 'https://github.com/SwaprHQ/presagio';
+
+const formatEventName = (eventName: string) =>
+  eventName.split(' ').join('-').toLocaleLowerCase();
+export const FA_EVENTS = {
+  FOOTER: {
+    CONNECT: {
+      DISCORD: 'click/footer-connect_discord',
```
Comment (berteotti) [inline constants/index.ts:89]: Why do we have dashes and underscores?

File: constants/index.ts [hunk 3]
```diff
@@ -76,3 +76,61 @@ export const DISCORD_URL =
   process.env.NEXT_PUBLIC_DISCORD_URL || 'https://discord.com/invite/QFkNsjTkzD';
 export const GITHUB_URL =
   process.env.NEXT_PUBLIC_GITHUB_URL || 'https://github.com/SwaprHQ/presagio';
+
+const formatEventName = (eventName: string) =>
+  eventName.split(' ').join('-').toLocaleLowerCase();
+export const FA_EVENTS = {
```
Comment (berteotti) [inline constants/index.ts:82]: Probably it's better to have a analytics file, right?

File: constants/index.ts (hunk 2 above)
Comment (ElRodrigote) [reply constants/index.ts:89]: I used them to express a hierarchy, in my head:

`$eventTrigger/$path-to-event_$event-detail`

File: constants/index.ts (hunk 3 above)
Comment (ElRodrigote) [reply constants/index.ts:82]: Yes! I agree, this `FA_EVENTS` object feels out of place now hahaha, I'll move it, maybe to its own module (`@/analytics`) or inside the `useAnalytics` hook file

File: app/(main)/markets/News.tsx [hunk 4]
```diff
@@ -47,11 +50,11 @@ export const News = ({ id }: NewsProps) => {
   if (!news) return <div className="py-4 text-center">No news</div>;
 
   return news.map((newsArticle, index) => (
-    <NewsArticle key={index} url={newsArticle.url} title={newsArticle.title} />
+    <NewsArticle id={id} key={index} url={newsArticle.url} title={newsArticle.title} />
   ));
 };
 
-const NewsArticle = ({ url, title }: NewsArticleProps) => {
+const NewsArticle = ({ id, url, title }: NewsArticleProps) => {
```
Comment (Diogomartf) [inline app/(main)/markets/News.tsx:57]: Don't think we're using id anymore
```suggestion
const NewsArticle = ({ url, title }: NewsArticleProps) => {
```

File: app/(main)/markets/News.tsx [hunk 5]
```diff
@@ -77,6 +80,7 @@ const NewsArticle = ({ url, title }: NewsArticleProps) => {
       href={url}
       target="_blank"
       rel="noopener noreferrer"
+      onClick={() => trackEvent(FA_EVENTS.MARKETS.DETAILS.TABS.NEWS_ARTICLE)}
```
Comment (Diogomartf) [inline app/(main)/markets/News.tsx:83]: we can pass an article title tho

File: app/(main)/markets/News.tsx [hunk 6]
```diff
@@ -47,11 +50,11 @@ export const News = ({ id }: NewsProps) => {
   if (!news) return <div className="py-4 text-center">No news</div>;
 
   return news.map((newsArticle, index) => (
-    <NewsArticle key={index} url={newsArticle.url} title={newsArticle.title} />
+    <NewsArticle id={id} key={index} url={newsArticle.url} title={newsArticle.title} />
```
Comment (Diogomartf) [inline app/(main)/markets/News.tsx:53]:
```suggestion
    <NewsArticle key={index} url={newsArticle.url} title={newsArticle.title} />
```

File: analytics/index.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    FILTERS: {
+      CREATOR: (creatorName: string) =>
+        `click/markets/filters/creator-${formatEventName(creatorName)}`,
+      ORDER: (orderName: string) =>
+        `click/markets/filters/order-${formatEventName(orderName)}`,
+      STATE: (stateName: string) =>
+        `click/markets/filters/state-${formatEventName(stateName)}`,
+      TOKEN: (tokenName: string) =>
+        `click/markets/filters/token-${formatEventName(tokenName)}`,
+    },
```
Comment (berteotti) [inline analytics/index.ts:59]: Maybe having a way to know what filters are applied on the url is better than tracking every click? Also, should we track every option click or just clicking on each filter select?

Comment (ElRodrigote) [reply analytics/index.ts:59]: I didn't find a way to track search params from the URL yet. That would solve the market details tracking (as we could use the `?id` from the URL)

If we want to know if the users are using filters, we can drop the option clicks.
If we want to know the most popular filters, the current approach works.

Follows a similar heuristic to the one used for the market categories: registering a click on the categories control vs registering every category click.

File: app/(main)/markets/News.tsx (hunk 5 above)
Comment (ElRodrigote) [reply app/(main)/markets/News.tsx:83]: Yes! We can. I'm not sure if it's worth it to track which title the user clicks, as each market will have its own articles, so there will be a lot of single click events in market news that are closed already

File: analytics/index.ts (hunk 7 above)
Comment (berteotti) [reply analytics/index.ts:59]: We can go forward with this solution

**Review by berteotti — APPROVED** [review]
Good job Rorry, looks good to me!

**Pull request merged**